packages/adblocker-webextension/adblocker.ts

```diff
@@ -505,6 +572,54 @@ export class WebExtensionBlocker extends FiltersEngine {
           },
     );
   }
+
+  private executeScriptlets(browser: Browser, tabId: number, scripts: string[]): void {
+    // Dynamically injected scripts scripts can be difficult to find later in
+    // the debugger. Console logs simplifies setting up breakpoints if needed.
+    let debugMarker;
+    if (this.config.debug) {
+      debugMarker = (text: string) =>
+        `console.log('[ADBLOCKER-DEBUG]:', ${JSON.stringify(text)});`;
+    } else {
+      debugMarker = () => '';
+    }
+
+    // the scriptlet code that contains patches for the website
+    const codeRunningInPage = `(function(){
+${debugMarker('run scriptlets (executing in "page world")')}
+${scripts.join('\n\n')}}
+)()`;
+
+    // wrapper to break the "isolated world" so that the patching operates
+    // on the website, not on the content script's isolated environment.
+    const codeRunningInContentScript = `
+(function(code) {
+    ${debugMarker('run injection wrapper (executing in "content script world")')}
+    var script;
+    try {
+      script = document.createElement('script');
+      script.appendChild(document.createTextNode(decodeURIComponent(code)));
+      (document.head || document.documentElement).appendChild(script);
+    } catch (ex) {
+      console.error('Failed to run script', ex);
+    }
+    if (script) {
+        if (script.parentNode) {
+          script.parentNode.removeChild(script);
+        }
+        script.textContent = '';
+    }
+})(\`${encodeURIComponent(codeRunningInPage)}\`);`;
```

Comment:
Why do we have to pass the code encoded as an argument instead of inlining it directly in the `document.createTextNode` call?

Reply:
The challenge to properly quote arbitrary code and pass it through two layers (background -> content script -> script element). Without additional knowledge on scriptlets themselves, I see no alternative to quoting one way or the other. encodeURIComponent is one way.

To illustrate the problem, passing it directly with something like that here

```
script.appendChild(document.createTextNode(\`${codeRunningInPage}\`));
``` 

... fails in the first scriptlets as `${}` is being used in a regular expression 
```
/[.*+?^${}()|[\]\\]/
```

In general, I see no option to avoid quoting (since the scriptlets can be any JavaScript code) but it is rather the question how to quote best.